apis/manifests/v1alpha1/packagemanifest_types.go

```diff
@@ -135,6 +146,28 @@ func (m *PackageManifest) Validate() error {
 		}
 	}
 
+	specImages := field.NewPath("spec").Child("images")
+	if len(m.Spec.Images) > 0 {
```

Comment:
Nit: You could remove the condition for this block and and always iterate over `m.Spec.Images`. Downside would be that the allocation of `existingNames` would sometimes be unnecessary but I would argue that is this a rather miniscule price to reduce the cyclo complexity here.

If you do not know yet, a nil slice is considered empty, you can iterate over it.

Reply:
Thx for the suggestion! Changed :+1: